### ISSUE TITLE
$rel may be null in Result.php

### ISSUE THREAD

**Issue by on-meetsys** [graphaware/neo4j-php-client#13]
Hello,

If I query nodes and relations in my neo4j db with '
`MATCH (p)-[r]->(q) RETURN *`
everything is fine. But I only get the linked nodes. 

So, I am trying to send a query like 
`MATCH (p) OPTIONAL MATCH (p)-[r]->(q) RETURN *`
to have all nodes and relations.

Unfortunately, then, some relations are null, and it makes an error on line 232 in Result.php, because $res is null.
I've therefore modified it to :
```
                if (null === $rel ||$rel->getId() === $relationshipId) {
                    return;
                }
```

And it works. I wonder if it is the right way to make it work.

Thanks.

**Comment by ikwattro** [comment]
Hi, thanks for the report, I will investigate that. If you're not too far in your dev, I would recommend to move to the 4.0 branch

**Comment by ikwattro** [comment]
Thanks again @on-meetsys . It has been fixed and released in 3.3.20
